apps/shipments/iot_client.py

```diff
@@ -36,3 +42,93 @@ def update_shadow(self, device_id, shadow):
             raise AWSIoTError("Error in response from AWS IoT")
 
         return iot_shadow['data']
+
+    def get_list_owner_devices(self, owner_id, next_token=None, results=None, **kwargs):
```

Comment:
In general, I think it's a good idea to avoid recursive functions - it sacrifices code readability for 'elegance'. I think it's fine to use a while loop. Something like a `while` loop that continues until next_token is falsy.

This change would also reduce the number of arguments to this function by 2 so you could remove the kwargs checks.